src/Controls/src/Core/Platform/Android/TabbedPageManager.cs

```diff
@@ -655,76 +655,70 @@ protected virtual ColorStateList GetItemTextColorStates()
 
 		protected virtual ColorStateList GetItemIconTintColorState()
 		{
-			if (IsBottomTabPlacement)
-			{
-				if (_orignalTabIconColors is null)
-					_orignalTabIconColors = _bottomNavigationView.ItemIconTintList;
-			}
-			// this ensures that existing behavior doesn't change
-			else if (!IsBottomTabPlacement && BarSelectedItemColor != null && BarItemColor == null)
-				return null;
+			if (_orignalTabIconColors is null)
+				_orignalTabIconColors = IsBottomTabPlacement ? _bottomNavigationView.ItemIconTintList : _tabLayout.TabIconTint;
 
 			Color barItemColor = BarItemColor;
 			Color barSelectedItemColor = BarSelectedItemColor;
 
-			if (barItemColor == null && barSelectedItemColor == null)
+			if (barItemColor is null && barSelectedItemColor is null)
 				return _orignalTabIconColors;
 
-			if (_newTabIconColors != null)
+			if (_newTabIconColors is not null)
 				return _newTabIconColors;
 
 			int defaultColor;
+			int checkedColor;
 
 			if (barItemColor is not null)
-			{
 				defaultColor = barItemColor.ToPlatform().ToArgb();
-			}
 			else
-			{
+				defaultColor = GetDefaultColor();
+
+			if (barSelectedItemColor is not null)
+				checkedColor = barSelectedItemColor.ToPlatform().ToArgb();
+			else
+				checkedColor = GetDefaultColor();
+
+			_newTabIconColors = GetColorStateList(defaultColor, checkedColor);
+			return _newTabIconColors;
+		}
+
+		int GetDefaultColor()
+		{
+			int defaultColor;
 #pragma warning disable XAOBS001 // Obsolete
```

Comment:
Since we are touching this code, can we check if theres a modern way (not obsolete) to improve getting the style attributes

Reply:
I have updated the code to retrieve `style attributes` using a modern approach without obsolete method. @rmarinho